cli/src/diff_util.rs

```diff
@@ -81,6 +87,8 @@ pub struct DiffFormatArgs {
     /// Generate diff by external command
     #[arg(long)]
     pub tool: Option<String>,
+    #[arg(long, requires("tool"))]
+    pub tool_mode: Option<DiffToolMode>,
```

Comment:
Optional: I'm a bit torn on whether this option is a good idea, but if we do it, how about:

```
jj diff --file-by-file-tool pratdiff
```

(or `--tool-file-by-file`; I also considered `--file-by-file --tool pratdiff`)

Note that this doesn't actually work very well: with this setup, `pratdiff` does not show color, and I couldn't find a way to fix it without creating a diff config for `pratdiff`. So, maybe just having a config know is sufficient.

Reply:
I removed the argument and made this a config only setting